### ISSUE TITLE
aXe Coconut issue for rule: Form elements must have labels

### ISSUE THREAD

**Issue by mfairchild365** [dequelabs/axe-core#1202]
It appears that the [hidden-explicit-label](https://github.com/dequelabs/axe-core/blob/master/lib/checks/label/hidden-explicit-label.json) rule was added a few months ago which will only pass if a "Form element has a visible explicit `<label>`"

I think that is a good addition, but I found a use case where this might result in a false positive.

For instance, the  [Material Angular autocomplete examples](https://material.angular.io/components/autocomplete/examples) result in this rule failing 5 times. Each field that fails this has an explcit label, but that label is absolutly positioned to help it pretend to be placeholder text.

This test starts passing when I use the dev tools in Chrome 70 to remove the `position: absolute` rule. That makes me wonder if there is a bug in [`dom.isOffscreen()`](https://github.com/dequelabs/axe-core/blob/master/lib/commons/dom/is-offscreen.js#L25) or something else. Resizing the browser window will affect wheather or not this error is reported. This error is not being reported in FF 63 as far as I can tell.

Thoughts?

**Comment by dylanb** [comment]
I think the bug here is that the label is showing up as hidden.

**Comment by ABUofA** [comment]
I just ran across this too, in regular Axe, both `axe-devtools 3.5.2 / axe-core 3.1.1` in Firefox 63 and `axe-devtools 3.6.0 / axe-core 3.1.1` in Chrome 70.

In our case, the CSS used to "hide" the explicitly-associated labels is almost line-for-line the same as the example given here - https://www.w3.org/WAI/tutorials/forms/labels/#note-on-hiding-elements

With that said, it has made us ask the question: why are we hiding these labels in the first place?

So something positive has come from this bug :+1:

**Comment by iamrafan** [comment]
I found one other false positive scenario where "hidden-explicit-label" check fails for an element whose explicit label is clipped and placeholder content is used as a visible label.

- Demo: https://codepen.io/iamrafan/pen/GwErRX?editors=1100#0

- Browser/AT experience :

  | AT/Browser | Announced as
-- | -- | --
1 | Edge/Narrator | Search editing
2 | Edge/JAWS | Search Edit
3 | Edge/NVDA | Search edit  Name-1 Name-1
4 | Chrome/NVDA | Search edit Name-1   blank
5 | Chrome/JAWS | Search Edit Name-1
6 | IE/JAWS | Search Edit
7 | IE/NVDA | Search edit blank
8 | Firefox/NVDA | Search edit  has auto complete blank

Changing the placeholder content here from "Name-1" to "Search" would provide the same experience for both Non AT and AT users.

Ex: 
```
  <label class="yj-acc-hidden" for="search">Search</label>
  <input type="text" placeholder="Search" id="search">
```

**Comment by WilcoFiers** [comment]
This problem has been resolved for axe-core 3.2.
